### DISCUSSION TITLE
Project directory structure design

### DISCUSSION THREAD

**Discussion by MaksymilianDemitraszek** [software-mansion/protostar#16] [General]
### Project directory structure overview
```
my-package/
├─ src/
│  ├─ main.cairo
├─ tests/
│  ├─ test.cairo
├─ package.toml
├─ README.md

```
### `package.toml` structure
```
[protostar.general]
name = "my-project"
version = "0.1.0"
description = "Some description"
authors = ["Adam Smith"]
license = "MIT"

[protostar.dependencies]
#TBD

[protostar.dev-dependencies]
#TBD


```

**Comment by kasperski95** [comment]
Each property in `package.toml` should have a justification why it is needed (what's the purpose of `readme`?).

Some properties may change due to package manager is mentioned as separate project:
https://starkware.notion.site/Cairo-Package-Manager-similar-to-Rust-Crate-e3f668cde90c4996afbd8af4b42bd9bf

**Comment by kasperski95** [comment]
I'd add a directory `lib` to the project directory structure. (I am not sure whether it should be named `lib` or `cairo_lib`).

**Reply by MaksymilianDemitraszek** [reply]
I think `lib` is good name. Although, I think it shouldn't be created with a project template, but when dependencies installed.
